pkg/render/component/component.go

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
```

Comment:
Thinking about this, I was hoping once everything in render is subpackaged that this would be one of the definitions that remains in the parent package. A common (and good) pattern is to have the interface definition in the parent package and the different implementations in the subpackages. The subpackages should import from the parent, and not the reverse, so there should never be circular imports.

If you've run into a circular import (i.e. the render package is importing from a subpackage) then I think something else needs to change.